examples/cifar10/train.cpp

```diff
@@ -168,9 +165,26 @@ int main(int argc, char **argv) {
               << "--minibatch_size 10 --backend_type internal" << std::endl;
     return -1;
   }
-  std::cout << "Running with learning rate " << learning_rate
-            << " and minibatch of size " << minibatch_size << " for " << epochs
-            << " epochs." << std::endl;
+  if (learning_rate <= 0) {
+    std::cerr << "Invalid learning rate. Learning rate must be greater than 0"
+              << std::endl;
+    return -1;
+  }
+  if (epochs <= 0) {
+    std::cerr << "Invalid epochs number. Epochs number must be greater than 0"
+              << std::endl;
+    return -1;
+  }
+  if (minibatch_size <= 0 || minibatch_size > 50000) {
+    std::cerr << "Invalid minibatch size. Minibatch rate must be greater than 0"
+                 " and less than dataset size (50000)"
+              << std::endl;
+    return -1;
+  }
+  std::cout << "Running with following parameters:" << std::endl
+            << "Learning rate: " << learning_rate << std::endl
+            << "Minibatch size: " << minibatch_size << std::endl
+            << "Epochs: " << epochs << std::endl;
```

Comment:
Please also print out `backend_type`. If it is printed in a log file, we can know which backend was used during training.